icu4c/source/i18n/erarules.cpp

```diff
@@ -116,16 +116,17 @@ EraRules* EraRules::createInstance(const char *calType, UBool includeTentativeEr
     ures_getByKey(rb.getAlias(), "calendarData", rb.getAlias(), &status);
     ures_getByKey(rb.getAlias(), calType, rb.getAlias(), &status);
     ures_getByKey(rb.getAlias(), "eras", rb.getAlias(), &status);
-        
+
     if (U_FAILURE(status)) {
         return nullptr;
     }
 
     int32_t numEras = ures_getSize(rb.getAlias());
     int32_t firstTentativeIdx = MAX_INT32;
 
-    LocalArray<int32_t> startDates(new int32_t[numEras], status);
-    if (U_FAILURE(status)) {
+    LocalMemory<int32_t> startDates(reinterpret_cast<int32_t *>(uprv_malloc(numEras * 4)));
```

Comment:
Should this not just be a static_cast<>? There should be no reinterpretation going on here, just casting a void* to the correct pointer type.

Reply:
You are right, done.